### PR TITLE
Ports TG Three view setup version 2

### PR DESCRIPTION
**Pull request by Aurorablade**
remake of https://github.com/ParadiseSS13/Paradise/pull/7820 cuase unsorted.dm is evil..in ways people do not realize.

Ports the three view character prefrence setup from TG, only without the image mapping bit

:cl: Fethas
tweak:Preference backend changed to TG style and to show more then one angle.
add:The tourist outfit actually exists now, not as an open player spawn but its no longer a overlay on top of ghosts only.
/:cl:

### CODE REVIEW

File: code/modules/client/preference/preferences.dm [hunk 1]
```diff
@@ -2050,7 +2047,7 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 
 				if("ghost_radio")
 					toggles ^= CHAT_GHOSTRADIO
-				
+
```
Comment (Aurorablade) [inline code/modules/client/preference/preferences.dm]: WHAT THE HELL ARE YOU

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 2]
```diff
@@ -417,7 +417,7 @@
 		if(2)	return FALSE //if it returns 2, it wants no normal handling
 	if(!has_organ_for_slot(slot))
 		return FALSE
-	
+
```
Comment (Aurorablade) [inline code/modules/mob/living/carbon/human/inventory.dm]: WHAT?!

File: code/modules/mob/new_player/preferences_setup.dm [hunk 3]
```diff
@@ -207,751 +207,73 @@
 			clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
 	return clothes_s
 
-/datum/preferences/proc/update_preview_icon(var/for_observer=0)		//seriously. This is horrendous.
```
Comment (Aurorablade) [inline code/modules/mob/new_player/preferences_setup.dm]: REST IN HELL

File: code/modules/client/preference/preferences.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/modules/client/preference/preferences.dm]: Probably your IDE cleaning up extra spaces

File: code/modules/mob/living/carbon/human/inventory.dm (hunk 2 above)
Comment (AffectedArc07) [reply code/modules/mob/living/carbon/human/inventory.dm]: Ditto

**Comment by Fox-McCloud** [comment]
@Aurorablade are there any issues/runtimes/problems with this?

File: code/modules/awaymissions/corpse.dm [hunk 4]
```diff
@@ -128,6 +128,7 @@
 	var/disable_pda = TRUE
 	var/disable_sensors = TRUE
 	//All of these only affect the ID that the outfit has placed in the ID slot
+	var/has_id = TRUE			//If the Mob actually has an ID, typically setting to false is gonna put the visual only equip flag to true. Prevent Mindless mob id runtimes.
```
Comment (Fox-McCloud) [inline code/modules/awaymissions/corpse.dm]: Why was this necessary?

File: code/modules/awaymissions/corpse.dm [hunk 5]
```diff
@@ -432,6 +436,7 @@
 
 /obj/effect/mob_spawn/human/miner/explorer
 	outfit = /datum/outfit/job/mining/equipped
+	has_id = FALSE //this spawns as a dead mob in a ruin, and makes a mindless mob runtime in current code.
```
Comment (Fox-McCloud) [inline code/modules/awaymissions/corpse.dm]: What was the runtime? THis is a hacky way of fixing it.

File: code/modules/client/preference/preferences.dm [hunk 6]
```diff
@@ -2114,7 +2111,7 @@ var/global/list/special_role_times = list( //minimum age (in days) for accounts
 
 /datum/preferences/proc/copy_to(mob/living/carbon/human/character)
 	var/datum/species/S = GLOB.all_species[species]
-	character.set_species(S.type) // Yell at me if this causes everything to melt
+	character.set_species(S.type, skip_same_check = TRUE) // Yell at me if this causes everything to melt
```
Comment (Fox-McCloud) [inline code/modules/client/preference/preferences.dm]: What issues was this causing?

File: code/__HELPERS/icons.dm [hunk 7]
```diff
@@ -960,4 +960,32 @@ The _flatIcons list is a cache for generated icon files.
...
+//For creating consistent icons for human looking simple animals
+/proc/get_flat_human_icon(icon_id, datum/job/J, datum/preferences/prefs, dummy_key, showDirs = cardinal, outfit_override = null)
+	var/static/list/humanoid_icon_cache = list()
+	if(!icon_id || !humanoid_icon_cache[icon_id])
+		var/mob/living/carbon/human/dummy/body = generate_or_wait_for_human_dummy(dummy_key)
+
+		if(prefs)
+			prefs.copy_to(body)
+		if(J)
+			J.equip(body, TRUE, FALSE)
```
Comment (Fox-McCloud) [inline code/__HELPERS/icons.dm]: You need the outfit_override argument here, and you'll need to change `/datum/job/equip` to have this:

```
	if(outfit_override || outfit)
		H.equipOutfit(outfit_override ? outfit_override : outfit, visualsOnly)
```

File: code/__HELPERS/icons.dm [hunk 8]
```diff
@@ -960,4 +960,32 @@ The _flatIcons list is a cache for generated icon files.
...
+		else if(outfit_override)
+			body.equipOutfit(outfit_override, visualsOnly = TRUE)
```
Comment (Fox-McCloud) [inline code/__HELPERS/icons.dm]: then this isn't necessary.

File: code/__HELPERS/icons.dm [hunk 9]
```diff
@@ -960,4 +960,32 @@ The _flatIcons list is a cache for generated icon files.
...
+
+
+
+		var/icon/out_icon = icon('icons/effects/effects.dmi', "nothing")
+
+		for(var/D in showDirs)
+			body.setDir(D)
```
Comment (Fox-McCloud) [inline code/__HELPERS/icons.dm]: no `COMPILE_OVERLAYS(body)`?

File: code/datums/datacore.dm [hunk 10]
```diff
@@ -1,5 +1,6 @@
 /hook/startup/proc/createDatacore()
 	data_core = new /datum/datacore()
+	var/static/list/show_directions = list(SOUTH, WEST)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: You have this twice; here, and..

File: code/datums/datacore.dm [hunk 11]
```diff
@@ -252,6 +253,7 @@ var/global/list/PDA_Manifest = list()
 
 var/record_id_num = 1001
 /datum/datacore/proc/manifest_inject(mob/living/carbon/human/H)
+	var/static/list/show_directions = list(SOUTH, WEST)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: Here.

File: code/modules/mob/living/carbon/human/dummy.dm [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	in_use = FALSE
+	return ..()
+
+/mob/living/carbon/human/dummy/Life()
+	return
+
+/mob/living/carbon/human/dummy/proc/wipe_state()
+	for(var/slot in get_all_slots())
+		qdel(slot)
+	overlays.Cut()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/carbon/human/dummy.dm]: we can use the` cut_overlays()` proc now.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 13]
```diff
@@ -207,751 +207,73 @@
...
-	preview_icon.Blend(face_s, ICON_OVERLAY)
-	preview_icon_front = new(preview_icon, dir = SOUTH)
-	preview_icon_side = new(preview_icon, dir = WEST)
-
-	qdel(face_s)
-	qdel(underwear_s)
-	qdel(undershirt_s)
-	qdel(socks_s)
-	qdel(clothes_s)
+				preview_icon = icon('icons/mob/robots.dmi', "robot", SOUTH)
```
Comment (Fox-McCloud) [inline code/modules/mob/new_player/preferences_setup.dm]: why not:

```
/datum/preferences/proc/update_preview_icon()
	// Determine what job is marked as 'High' priority, and dress them up as such.
	var/datum/job/previewJob
	var/highest_pref = 0
	for(var/job in job_preferences)
		if(job_preferences[job] > highest_pref)
			previewJob = SSjob.GetJob(job)
			highest_pref = job_preferences[job]

	if(previewJob)
		// Silicons only need a very basic preview since there is no customization for them.
		if(istype(previewJob,/datum/job/ai))
			parent.show_character_previews(image('icons/mob/ai.dmi', icon_state = resolve_ai_icon(preferred_ai_core_display), dir = SOUTH))
			return
		if(istype(previewJob,/datum/job/cyborg))
			parent.show_character_previews(image('icons/mob/robots.dmi', icon_state = "robot", dir = SOUTH))
			return

	// Set up the dummy for its photoshoot
	var/mob/living/carbon/human/dummy/mannequin = generate_or_wait_for_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)
	copy_to(mannequin)

	if(previewJob)
		mannequin.job = previewJob.title
		previewJob.equip(mannequin, TRUE, preference_source = parent)

	COMPILE_OVERLAYS(mannequin)
	parent.show_character_previews(new /mutable_appearance(mannequin))
	unset_busy_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)
```
?

File: code/game/jobs/job/security.dm [hunk 14]
```diff
@@ -136,13 +136,13 @@
...
-/datum/outfit/job/detective/pre_equip(mob/living/carbon/human/H, visualsOnly = FALSE)
+/datum/outfit/job/detective/pre_equip(mob/living/carbon/human/H, visualsOnly = FALSE, alt_title)
 	. = ..()
-	if(H.mind && H.mind.role_alt_title)
-		switch(H.mind.role_alt_title)
-			if("Forensic Technician")
-				suit = /obj/item/clothing/suit/storage/det_suit/forensics/blue
-				head = null
+	if(!alt_title)
+		alt_title = H.mind && H.mind.role_alt_title ? H.mind.role_alt_title : null
```
Comment (Fox-McCloud) [inline code/game/jobs/job/security.dm]: Why have these two lines above?

Comment (Fox-McCloud) [inline code/game/jobs/job/security.dm]: I'd just can the other two lines above, to be honest.

File: code/game/jobs/job/security.dm [hunk 15]
```diff
@@ -136,13 +136,13 @@
...
+	if(alt_title == "Forensic Technician" )
```
Comment (Fox-McCloud) [inline code/game/jobs/job/security.dm]: don't forget the medical doctor and psych, who have alternative outfits.

File: code/datums/datacore.dm [hunk 16]
```diff
@@ -281,7 +283,7 @@ var/record_id_num = 1001
 		G.fields["m_stat"]		= "Stable"
 		G.fields["sex"]			= capitalize(H.gender)
 		G.fields["species"]		= H.dna.species.name
-		G.fields["photo"]		= get_id_photo(H)
+		G.fields["photo"]		= get_id_photo(H, show_directions)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: these arguments

File: code/datums/datacore.dm [hunk 17]
```diff
@@ -343,310 +345,11 @@ var/record_id_num = 1001
...
-		if(H.dna.species.bodyflags & HAS_SKIN_COLOR)
-			temp.Blend(H.skin_colour, H.body_accessory.blend_mode)
-		if(t_marking_s)
-			temp.Blend(t_marking_s, ICON_OVERLAY)
-		preview_icon.Blend(temp, ICON_OVERLAY)
-	qdel(face_s)
-	qdel(clothes_s)
-
-	return preview_icon
+/proc/get_id_photo(mob/living/carbon/human/H, client/C, show_directions = list(SOUTH))
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: Don't match up with these arguments.

File: code/modules/awaymissions/corpse.dm [hunk 18]
```diff
@@ -205,7 +206,10 @@
 			var/T = vars[slot]
 			if(!isnum(T))
 				outfit.vars[slot] = T
-		H.equipOutfit(outfit)
```
Comment (Fox-McCloud) [inline code/modules/awaymissions/corpse.dm]: Keep this, and let's solve the runtime.

File: code/modules/client/preference/preferences.dm (hunk 1 above)
Comment (Aurorablade) [reply code/modules/client/preference/preferences.dm]: I use DM

File: code/__HELPERS/icons.dm (hunk 9 above)
Comment (Aurorablade) [reply code/__HELPERS/icons.dm]: won't compile for some reason https://puu.sh/EcMeZ/0f81724b9c.png

**Comment by Aurorablade** [comment]
Posting changes and leaving note on PR for future feedback/review/Fox shotting me with a gun:
current runtimes: Mindless mobs corpse spawners:
`[2019-09-02T23:33:16] Runtime in datacore.dm,350: Cannot read null.assigned_role
  proc name: get id photo (/proc/get_id_photo)
  src: null
  call stack:
  get id photo(Jorge Ratcliff (/mob/living/carbon/human), null, /list (/list))
  Clown (/datum/outfit/job/clown): imprint idcard(Jorge Ratcliff (/mob/living/carbon/human))
  Clown (/datum/outfit/job/clown): post equip(Jorge Ratcliff (/mob/living/carbon/human), 0)
  Clown (/datum/outfit/job/clown): post equip(Jorge Ratcliff (/mob/living/carbon/human), 0)
  Clown (/datum/outfit/job/clown): equip(Jorge Ratcliff (/mob/living/carbon/human), 0, null)
  Jorge Ratcliff (/mob/living/carbon/human): equipOutfit(Clown (/datum/outfit/job/clown), 0, null)
  Clown (/obj/effect/mob_spawn/human/clown/corpse): equip(Jorge Ratcliff (/mob/living/carbon/human), 1)
  Clown (/obj/effect/mob_spawn/human/clown/corpse): create(null, 1, null)
  Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1)
  Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1)
  Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1)
  Atoms (/datum/controller/subsystem/atoms): InitAtom(Clown (/obj/effect/mob_spawn/human/clown/corpse), /list (/list))
  Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null, 1)
  Atoms (/datum/controller/subsystem/atoms): Initialize(127864)
  Master (/datum/controller/master): Initialize(10, 0)`
Proc suggetion for /datum/preferences/proc/update_preview_icon():
Won't work at this time due to the fact our job prefrenence stuff is controlled by a GetJobDepartment Proc SWITCH (https://github.com/ParadiseSS13/Paradise/blob/fc83499392a83aae858a3764e123179ae111eb9c/code/controllers/subsystem/jobs.dm#L144) Vs TGs Job Prefrences list. I am not sure how to cleanly replicate it without a full on prefrence overhaul (not to be seen in this PR). I did run a test of it trying ssjobs.Occupations and it didn't work how i was doing it. Open to suggestions. Though probaly need to slap outfit overides in correct spots.

Oddity: surgery bluecap not showing up surgeon in preview...possibly how i test local.

Error: Photos not showing up, though i feel its how i am testing on local. May need someone else to back check.

Adding compile_overlays(body):That runtimed on me..said lazylen was not defined. Not sure if due to the nature of how the proc is..or what.

Observertions: Our prefrences need an overhual and out photocode procs DAMN SURE need one.

File: code/modules/client/preference/preferences.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/modules/client/preference/preferences.dm]: Can I recommend VSCode? It has:
- Global search across workspace
- Allows you to jump to proc and bar definitions even if in separate files
- Has built in git so you don’t need another client
- Dark Mode

That’s just the surface

*100% not a paid representative*

File: code/datums/datacore.dm [hunk 19]
```diff
@@ -252,7 +251,7 @@ var/global/list/PDA_Manifest = list()
 		foundrecord.fields["real_rank"] = real_title
 
 var/record_id_num = 1001
-/datum/datacore/proc/manifest_inject(mob/living/carbon/human/H)
+/datum/datacore/proc/manifest_inject(mob/living/carbon/human/H, client/C)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: Oh, I wasn't suggesting adding this argument. our `manifest_inject` is completely different.

File: code/datums/datacore.dm [hunk 20]
```diff
@@ -283,7 +284,7 @@ var/record_id_num = 1001
 		G.fields["m_stat"]		= "Stable"
 		G.fields["sex"]			= capitalize(H.gender)
 		G.fields["species"]		= H.dna.species.name
-		G.fields["photo"]		= get_id_photo(H, show_directions)
+		G.fields["photo"]		= get_id_photo(H, C, show_directions)
```
Comment (Fox-McCloud) [inline code/datums/datacore.dm]: I was recommending you change the `get_id_photo` proc itself.

File: code/__HELPERS/icons.dm (hunk 9 above)
Comment (Fox-McCloud) [reply code/__HELPERS/icons.dm]: You need to update to a version of 512 or something. I recommend 512.1472

Comment (Fox-McCloud) [reply code/__HELPERS/icons.dm]: rename `__HELPERS/lists.dm` to `__HELPERS/_lists.dm

File: code/modules/awaymissions/corpse.dm [hunk 21]
```diff
@@ -206,9 +205,6 @@
 			var/T = vars[slot]
 			if(!isnum(T))
 				outfit.vars[slot] = T
-		if(!has_id)
-			H.equipOutfit(outfit, TRUE)
-		else
 			H.equipOutfit(outfit)
```
Comment (Fox-McCloud) [inline code/modules/awaymissions/corpse.dm]: YOu need to move this back....or else it's going to equip the outfit multiple times for all your slots.

File: code/datums/datacore.dm (hunk 17 above)
Comment (Fox-McCloud) [reply code/datums/datacore.dm]: I was recommending you change the get_id_photo proc itself.

File: code/modules/mob/new_player/preferences_setup.dm [hunk 22]
```diff
@@ -207,751 +207,73 @@
...
-						clothes_s.Blend(new /icon('icons/mob/back.dmi', "satchel"), ICON_OVERLAY)
-
-	else if(job_medsci_high)
-		switch(job_medsci_high)
-			if(RD)
-				clothes_s = new /icon(uniform_dmi, "director_s")
-				clothes_s.Blend(new /icon('icons/mob/feet.dmi', "brown"), ICON_UNDERLAY)
-				clothes_s.Blend(new /icon('icons/mob/suit.dmi', "labcoat_open"), ICON_OVERLAY)
-				if(prob(1))
-					clothes_s.Blend(new /icon('icons/mob/head.dmi', "petehat"), ICON_OVERLAY)
```
Comment (Fox-McCloud) [inline code/modules/mob/new_player/preferences_setup.dm]: these little rare things are the only thing I'll miss.

I once had a metroid on my head as a scientist---a bear belt, as a shaft miner, another time.

Comment (KasparoVy) [reply code/modules/mob/new_player/preferences_setup.dm]: Yeah when I saw that shit I thought it was a bug and had a good laugh. RIP!

File: code/modules/awaymissions/corpse.dm (hunk 21 above)
Comment (Aurorablade) [reply code/modules/awaymissions/corpse.dm]: Accured tabbing!

File: code/modules/mob/new_player/preferences_setup.dm (hunk 22 above)
Comment (Aurorablade) [reply code/modules/mob/new_player/preferences_setup.dm]: Anyway we could add it back somehow?

Comment (KasparoVy) [reply code/modules/mob/new_player/preferences_setup.dm]: Could probably factor in 'chance to assign fluff kit' into outfitting, with a list of defined object datums and their chances for use in `prob()`

Comment (Fox-McCloud) [reply code/modules/mob/new_player/preferences_setup.dm]: For now, I wouldn't worry about it; let's get this system, as is, in and functioning, since it's gone long enough without being merged and is already giving us enough trouble.

That said, it being re-added in the future, if it can be done in a non-snowflakey way (that'll be hard) would be great.

Comment (Aurorablade) [reply code/modules/mob/new_player/preferences_setup.dm]: agreed. just got to figure out the other stuff. But come as is...Hopeffully this big wet windstorm off the coast dosen't give me a time.

File: code/__HELPERS/icons.dm (hunk 9 above)
Comment (Fox-McCloud) [reply code/__HELPERS/icons.dm]: @Aurorablade this is still relevant and will allow you to use the COMPILE_OVERLAYS thing

Comment (Aurorablade) [reply code/__HELPERS/icons.dm]: yeah get on it ain a bit my email did not notify me i had a new comment..

**Comment by Aurorablade** [comment]
Conflicts noted, Will fix later kinda want to do it the same time i have a good idea how to fix that mindless mob issue.

**Comment by Aurorablade** [comment]
as a reminder of outstanding runtimes: 
current runtimes: Mindless mobs corpse spawners:
`[2019-09-02T23:33:16] Runtime in datacore.dm,350: Cannot read null.assigned_role proc name: get id photo (/proc/get_id_photo) src: null call stack: get id photo(Jorge Ratcliff (/mob/living/carbon/human), null, /list (/list)) Clown (/datum/outfit/job/clown): imprint idcard(Jorge Ratcliff (/mob/living/carbon/human)) Clown (/datum/outfit/job/clown): post equip(Jorge Ratcliff (/mob/living/carbon/human), 0) Clown (/datum/outfit/job/clown): post equip(Jorge Ratcliff (/mob/living/carbon/human), 0) Clown (/datum/outfit/job/clown): equip(Jorge Ratcliff (/mob/living/carbon/human), 0, null) Jorge Ratcliff (/mob/living/carbon/human): equipOutfit(Clown (/datum/outfit/job/clown), 0, null) Clown (/obj/effect/mob_spawn/human/clown/corpse): equip(Jorge Ratcliff (/mob/living/carbon/human), 1) Clown (/obj/effect/mob_spawn/human/clown/corpse): create(null, 1, null) Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1) Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1) Clown (/obj/effect/mob_spawn/human/clown/corpse): Initialize(1) Atoms (/datum/controller/subsystem/atoms): InitAtom(Clown (/obj/effect/mob_spawn/human/clown/corpse), /list (/list)) Atoms (/datum/controller/subsystem/atoms): InitializeAtoms(null, 1) Atoms (/datum/controller/subsystem/atoms): Initialize(127864) Master (/datum/controller/master): Initialize(10, 0)`

The suggestion of: `/datum/preferences/proc/update_preview_icon()
	// Determine what job is marked as 'High' priority, and dress them up as such.
	var/datum/job/previewJob
	var/highest_pref = 0
	for(var/job in job_preferences)
		if(job_preferences[job] > highest_pref)
			previewJob = SSjob.GetJob(job)
			highest_pref = job_preferences[job]

	if(previewJob)
		// Silicons only need a very basic preview since there is no customization for them.
		if(istype(previewJob,/datum/job/ai))
			parent.show_character_previews(image('icons/mob/ai.dmi', icon_state = resolve_ai_icon(preferred_ai_core_display), dir = SOUTH))
			return
		if(istype(previewJob,/datum/job/cyborg))
			parent.show_character_previews(image('icons/mob/robots.dmi', icon_state = "robot", dir = SOUTH))
			return

	// Set up the dummy for its photoshoot
	var/mob/living/carbon/human/dummy/mannequin = generate_or_wait_for_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)
	copy_to(mannequin)

	if(previewJob)
		mannequin.job = previewJob.title
		previewJob.equip(mannequin, TRUE)

	COMPILE_OVERLAYS(mannequin)
	parent.show_character_previews(new /mutable_appearance(mannequin))
	unset_busy_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)`

What was actually stopping me here is 'job_preferences' as thats a list from TG that i am unsure definitaly what is on our code ATM.

I can do some local testing better with my DB set up now.

File: code/__HELPERS/icons.dm [hunk 23]
```diff
@@ -960,4 +960,31 @@ The _flatIcons list is a cache for generated icon files.
...
+		if(J)
+			J.equip(body, TRUE, FALSE, outfit_override = outfit_override)
+		else if (outfit_override)
+			body.equipOutfit(outfit_override, visualsOnly = TRUE)
+
+
+		var/icon/out_icon = icon('icons/effects/effects.dmi', "nothing")
+		for(var/D in showDirs)
+			body.setDir(D)
+			COMPILE_OVERLAYS(body)
```
Comment (Aurorablade) [inline code/__HELPERS/icons.dm]: Theres that compile over lays fix.

File: code/__HELPERS/_lists.dm [hunk 24]
```diff
@@ -1,827 +1,827 @@
...
-		. = L
-
-/proc/counterlist_combine(list/L1, list/L2)
-	for(var/key in L2)
-		var/other_value = L2[key]
-		if(key in L1)
-			L1[key] += other_value
-		else
+/*
+ * Holds procs to help with list operations
```
Comment (Aurorablade) [inline code/__HELPERS/_lists.dm]: did it once AGAIN take an entire file when i did a upstream pull and mark it as the whole thing changed?
WHY
DOES
IT
DO THIS

File: code/game/jobs/job/job.dm [hunk 25]
```diff
@@ -248,10 +248,10 @@
 		C.sex = capitalize(H.gender)
 		C.age = H.age
 		C.name = "[C.registered_name]'s ID Card ([C.assignment])"
-		C.photo = get_id_photo(H)
-
-		if(H.mind && H.mind.initial_account)
-			C.associated_account_number = H.mind.initial_account.account_number
+		if(H.mind)//avoid mindless mob errors with these procs
```
Comment (Aurorablade) [inline code/game/jobs/job/job.dm]: so pretty much i looked in imprint ID (We have TG dosen't causes runtimes on our end) decided just to split up and scoot these over into a general mind check. seems to fix the issue.

File: code/__HELPERS/_lists.dm (hunk 24 above)
Comment (Aurorablade) [reply code/__HELPERS/_lists.dm]: Don't even know what commit did it either..

**Comment by Aurorablade** [comment]
also rolling back to something but the suggested bit here `/datum/preferences/proc/update_preview_icon()
	// Determine what job is marked as 'High' priority, and dress them up as such.
	var/datum/job/previewJob
	var/highest_pref = 0
	for(var/job in job_preferences)
		if(job_preferences[job] > highest_pref)
			previewJob = SSjob.GetJob(job)
			highest_pref = job_preferences[job]

	if(previewJob)
		// Silicons only need a very basic preview since there is no customization for them.
		if(istype(previewJob,/datum/job/ai))
			parent.show_character_previews(image('icons/mob/ai.dmi', icon_state = resolve_ai_icon(preferred_ai_core_display), dir = SOUTH))
			return
		if(istype(previewJob,/datum/job/cyborg))
			parent.show_character_previews(image('icons/mob/robots.dmi', icon_state = "robot", dir = SOUTH))
			return

	// Set up the dummy for its photoshoot
	var/mob/living/carbon/human/dummy/mannequin = generate_or_wait_for_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)
	copy_to(mannequin)

	if(previewJob)
		mannequin.job = previewJob.title
		previewJob.equip(mannequin, TRUE)

	COMPILE_OVERLAYS(mannequin)
	parent.show_character_previews(new /mutable_appearance(mannequin))
	unset_busy_human_dummy(DUMMY_HUMAN_SLOT_PREFERENCES)`

is 
` job_preferences`

In our variant we kinda do not have that or an equivalent to that right now, I SUSPECT it may have to do in TG preferences setup code wise. I tried a few different other lists we DO have and nothing kinda equivalent to.

**Comment by Aurorablade** [comment]
if i can figure out why is showing that many changes on the list.dm file i may have to remake this again.

**Comment by Aurorablade** [comment]
okay so that is what is happening..what it looks like is that the list file went to the unix format line endings at some point..and if autocrlf is on, git is gonna convert it over... ATM though it looks like both the line endings on the repo and my file are BOTH CRLF endings. So still dunno what witchery is going on. i can and will remake it if i have to. Question is should i turn that autocrlf off...Best practice feels like i should not.

**Comment by Aurorablade** [comment]
so can't fix the issue with it despite a few attempts will have to do version 3 of this later when my computer is back

**Comment by Aurorablade** [comment]
i am probaly gonnah have to do thins in a new branch later.

**Comment by AffectedArc07** [comment]
You dont need a new branch to fix the issues

Just stick new lines at the end of these
```
file ./code/modules/clothing/under/miscellaneous.dm is missing a trailing newline
file ./code/modules/mob/living/carbon/human/dummy.dm is missing a trailing newline
file ./code/__HELPERS/unsorted.dm is missing a trailing newline
file ./code/__HELPERS/icons.dm is missing a trailing newline
```

And change your line endings for these files in your IDE for these
```
code/modules/clothing/under/miscellaneous.dm
code/modules/mob/new_player/new_player.dm
code/modules/mob/new_player/preferences_setup.dm
code/modules/mob/dead/observer/observer.dm
code/modules/mob/living/carbon/human/human.dm
code/modules/mob/living/carbon/human/inventory.dm
code/datums/datacore.dm
code/game/jobs/job/job.dm
code/game/jobs/job/medical.dm
code/__HELPERS/icons.dm
```

In vscode, its done through a button in the bottom right
image

(This will say CRLF for you)

If you need any help just ask on discord

**Comment by Aurorablade** [comment]
well its also showing a bunch of files as a full edit that shouldn't be showing as a full edit. and that makes fox strangle me.

**Comment by Aurorablade** [comment]
but i will look at it in a bit see if i can do it..i see what your saying now.

**Comment by Aurorablade** [comment]
fix that other thing..laterish...

**Comment by Aurorablade** [comment]
i think i am as far as i can go on this. Unless someone slaps me with a suggestion. Some previous suggestions rely on some job SS code updates so we can easily loop through jobs better for checks.

**Comment by Aurorablade** [comment]
i promise i am going to finish this...Just covid and crazy retail in my area,

**Comment by KasparoVy** [comment]
Aurora you're a legend for trucking away on this still

**Comment by Aurorablade** [comment]
I am closing this cause i have had it open way to long and will probaly need to redo alot of stuff later. but i can do that. I am still around just IRL and retail kinda leaves me tired. I am coming back to this i just need to get my stuff together! Much love.
